minoc/src/intTest/java/org/opencadc/minoc/SetStateTest.java

```diff
@@ -0,0 +1,397 @@
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.minoc;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.auth.RunnableAction;
+import ca.nrc.cadc.net.HttpDelete;
+import ca.nrc.cadc.net.HttpGet;
+import ca.nrc.cadc.net.HttpPost;
+import ca.nrc.cadc.net.HttpTransfer;
+import ca.nrc.cadc.net.HttpUpload;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import ca.nrc.cadc.rest.RestAction;
+import ca.nrc.cadc.util.Log4jInit;
+
+import static org.junit.Assert.fail;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URL;
+import java.security.PrivilegedExceptionAction;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.security.auth.Subject;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test PUT, GET, DELETE, POST, HEAD and common errors
+ * Note: This test the minoc-test.pem cert and requires minoc-availability.properties 
+ *       to be in minoc/config/. The following entry needs to be in the 
+ *       minoc-availability.properties file.
+ *       users=CN=cadcauthtest1_24c, OU=cadc, O=hia, C=ca
+ * 
+ * @author yeunga
+ */
+public class SetStateTest extends MinocTest {
+    
+    private static final Logger log = Logger.getLogger(SetStateTest.class);
+    
+    static {
+        Log4jInit.setLevel("org.opencadc.minoc", Level.INFO);
+    }
+    
+    public SetStateTest() {
+        super();
+    }
+    
+    @Test
+    public void testSetOffline() throws Exception {
+        try {
+            // set minoc state to 'Offline'
```

Comment:
Test needs to confirm the initial state before testing that changes work; if it isn't in the right state to start (because another test left in the wrong state) then set it and confirm.

this comment applies to other test methods as well.

Reply:
Fixed.